packages/@aws-cdk/aws-ec2/README.md

```diff
@@ -608,6 +608,88 @@ new ec2.Instance(this, 'Instance', {
 
 ```
 
+## Volumes
+
+Whereas Block Devices are created, and automatically associated, with a specific instance when that instance is created. Volumes allow you to create a block device that can be attached to, or detached from, any instance at any time. Some types of Volumes can also be attached to multiple instances at the same time to allow you to have shared storage between those instances. A notable restriction is that a Volume can only be attached to instances in the same availability zone as the Volume itself.
+
+The following demonstrates how to create a 500 GiB encrypted Volume in the `us-west-2a` availability zone, and give a role the ability to attach that Volume to a specific instance:
+
+```ts
+const instance = new ec2.Instance(this, 'Instance', {
+  // ...
+});
+const role = new iam.Role(stack, 'SomeRole', {
+  assumedBy: new iam.AccountRootPrincipal(),
+});
+const volume = new ec2.Volume(this, 'Volume', {
+  availabilityZone: 'us-west-2a',
+  size: cdk.Size.gibibytes(500),
+  encrypted: true,
+});
+
+volume.grantAttachVolume(role, [instance]);
+```
+
+### Instances Attaching Volumes to Themselves
+
+If you need to grant an instance the ability to attach/detach an EBS volume to/from itself, then using `grantAttachVolume` and `grantDetachVolume` as outlined above
+will lead to an unresolvable circular reference between the instance role and the instance. To securely provide this grant we recommend conditioning the grant
```

Comment:
This shouldn't actually be true. The dependency graph could/should be looking like this:

```
┌─────────────┐               ┌─────────────┐
│             │               │             │
│  Instance   │──────────────▶│    Role     │
│             │               │             │
└─────────────┘               └─────────────┘
       ▲                             ▲       
       │                             │       
       │                             │       
       │                             │       
       │       ┌─────────────┐       │       
       │       │             │       │       
       └───────│   Policy    │───────┘       
               │             │               
               └─────────────┘               
```

Which should render fine.

What is the cause of the cyclical reference?

Reply:
Clarified off github.